### ISSUE TITLE
REST handler should detect protocol, host, port automatically for the endpoint

### ISSUE THREAD

**Issue by vbg8exadel** [zenstackhq/zenstack#737]
**Description and expected behavior**
As per documentation in order to use REST Handler we need to provide endpoint URL. Unfortunately the only possible format is absolute URL with hardcoded protocol, hostname and port number. See the documentation at https://zenstack.dev/docs/reference/server-adapters/api-handlers/rest 

```
handler: RestApiHandler({ endpoint: 'http://myhost/api' }) 
```

While testing locally it is enough to have endpoint configured as so, but once you start deploying the same code to multiple environments, then it becomes invalid. The protocol, host, port for the endpoint URL should be detected automatically so we can add the URL context only. For example, like so:

```
handler: RestApiHandler({ endpoint: '/api' }) 
```

After that I can run the server locally at http://localhost:3000, or deploy to DEV server which runs at https://dev-api.myhost.org:8080: my code will work properly, and the URLs will be pointing to correct place.

**Environment (please complete the following information):**

- ZenStack version: [e.g., 1.0.2]
- 

**Additional context**

**Comment by ymc9** [comment]
Hi @vbg8exadel , sorry for the late response. I understand it's not convenient. One possible solution is to put the deployment URI into an environment variable and use it to configure the API handler. IIRC, the full URL is required by JSON:API specification.

Maybe we can improve this by inspecting the request object and extracting the current base URL out of it.

**Comment by vbg8exadel** [comment]
Yes, that is one option. Although there should be a way to ger it from Express router. I hope there is a way to get the base URL once in the beginning.
